main.go

```diff
@@ -105,45 +105,53 @@ func main() {
 						exit(2)
 					} else {
 
-						for _, toPrune := range options.Merge.Prune {
-							DEBUG("%s: Pruning", toPrune)
-							keys := strings.Split(toPrune, ".")
-							if len(keys) > 1 {
-								parent := strings.Join(keys[0:(len(keys)-1)], ".")
-								target := keys[(len(keys) - 1)]
-								node, _ := resolveNode(parent, root)
-								if _, ok := node.(map[interface{}]interface{}); ok {
-									DEBUG("   PRUNED %q from %q", target, parent)
-									delete(node.(map[interface{}]interface{}), target)
-								}
-							} else {
-								DEBUG("   PRUNED %q from \"$\"", keys[0])
-								delete(root, keys[0])
-							}
-						}
-
-						over := ParamChecker{}
-						err = walkTree(root, over, "")
+						sref := StringReferencer{root: root}
+						err = walkTree(root, sref, "")
```

Comment:
Needs testing to ensure the errors from walkTree(root,sref,"") are getting through to the end user